indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -182,9 +216,11 @@ public TaskLock lock(final Task task, final Interval interval) throws Interrupte
     try {
       Optional<TaskLock> taskLock;
       while (!(taskLock = tryLock(task, interval)).isPresent()) {
+        tasksWaitingForLock.add(task);
+        log.info("Task [%s] added to list of tasks waiting for a lock release", task.getId());
         lockReleaseCondition.await();
       }
-
+      tasksWaitingForLock.remove(task);
```

Comment:
Should this be in a `try/finally`? It's possible for `lockReleaseCondition.await()` or `tryLock` to throw exceptions while a task is waiting, and in that case `tasksWaitingForLock` won't get updated. I _think_ the task should still get removed.